src/SourceLocator/Type/FileSystemIteratorSourceLocator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace BetterReflection\SourceLocator\Type;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+
+/**
+ * This source locator loads all php files from \FileSystemIterator
+ */
+class FileSystemIteratorSourceLocator implements SourceLocator
+{
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $_aggregatedSourceLocator;
+
+    /**
+     * @var \FilesystemIterator
+     */
+    private $fileSystemIterator;
+
+    public function __construct(\FilesystemIterator $filesystemIterator)
+    {
+        $this->fileSystemIterator = $filesystemIterator;
+    }
+
+    private function getAggregatedSourceLocator()
+    {
+        if (null==$this->_aggregatedSourceLocator) {
+            $sourceLocators = $this->scan();
```

Comment:
return `$this->aggregatedSourceLocator ?: $this->aggregatedSourceLocator = new new AggregateSourceLocator($this->scan());`